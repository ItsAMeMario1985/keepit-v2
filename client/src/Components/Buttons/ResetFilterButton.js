import PropTypes from 'prop-types'
import styled from 'styled-components/macro'
import { CloseIcon } from '../Icons'

ResetFilterButton.propTypes = {
  onClick: PropTypes.func.isRequired,
  buttonText: PropTypes.string,
  className: PropTypes.string,
}

export default function ResetFilterButton({ onClick, buttonText, className }) {
  return (
    <StyledResetFilterButton
      data-testid="resetbutton"
      className={className}
      onClick={onClick}
    >
      {buttonText} <CloseIcon />
    </StyledResetFilterButton>
  )
}

const StyledResetFilterButton = styled.button`
  border: none;
  outline: none;
  background-color: #b5b5b5;
  color: white;
  border-bottom-right-radius: 8px;
  border-top-right-radius: 8px;
  min-width: 50px;
  margin-top: 5px;
  align-items: center;
  padding: 5px 10px 5px 10px;
  border-left: 2px solid #e3e3e3;
  cursor: pointer;
  font-weight: 400;
  font-size: 14px;

  svg {
    margin-bottom: -2px;
  }
`
