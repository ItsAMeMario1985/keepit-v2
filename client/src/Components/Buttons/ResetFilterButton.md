```jsx
<ResetFilterButton buttonText="test" onClick={() => {}} />
```
