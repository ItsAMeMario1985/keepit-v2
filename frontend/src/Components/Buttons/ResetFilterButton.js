import PropTypes from 'prop-types'
import styled from 'styled-components/macro'
import { CloseIcon } from '../Icons'
export default function ResetFilterButton({ onClick, buttonText, className }) {
  return (
    <StyledResetFilterButton className={className} onClick={onClick}>
      {buttonText} <CloseIcon />
    </StyledResetFilterButton>
  )
}
const StyledResetFilterButton = styled.button`
  border: none;
  outline: none;
  background-color: #b5b5b5;
  color: white;
  border-bottom-right-radius: 8px;
  border-top-right-radius: 8px;
  min-width: 50px;
  margin-top: 5px;
  align-items: center;
  padding: 5px 10px 5px 10px;
  border-left: 2px solid #e3e3e3;
  cursor: pointer;
  font-weight: 400;
  font-size: 14px;
  margin-bottom: 90px;
  svg {
    margin-bottom: -2px;
  }
`
