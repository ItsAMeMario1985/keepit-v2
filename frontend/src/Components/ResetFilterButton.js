import PropTypes from 'prop-types'
import styled from 'styled-components/macro'
import { ReactComponent as FilterCloseIcon } from '../Assets/close.svg'

export default function ResetFilterButton({ onClick, buttonText, className }) {
  return (
    <StyledResetFilterButton className={className} onClick={onClick}>
      {buttonText} <FilterCloseIcon />
    </StyledResetFilterButton>
  )
}
const StyledResetFilterButton = styled.button`
  border: none;
  outline: none;
  background-color: #b5b5b5;
  color: white;
  border-bottom-right-radius: 8px;
  border-top-right-radius: 8px;
  min-width: 50px;
  margin-top: 5px;
  align-items: center;
  padding: 5px 10px 5px 10px;
  border-left: 2px solid #e3e3e3;
  cursor: pointer;
  font-weight: 400;
  font-size: 14px;

  svg {
    margin-bottom: -2px;
  }
`

/*


*/
