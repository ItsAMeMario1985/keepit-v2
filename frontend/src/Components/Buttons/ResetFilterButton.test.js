import { render } from '@testing-library/react'
import ResetFilterButton from './ResetFilterButton'
describe('ResetFilterButton', () => {
  it('renders correctly', () => {
    const { container } = render(<ResetFilterButton />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
