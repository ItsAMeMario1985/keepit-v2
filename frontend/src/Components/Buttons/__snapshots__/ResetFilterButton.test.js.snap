// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetFilterButton renders correctly 1`] = `
<button
  class="ResetFilterButton__StyledResetFilterButton-sc-12zadcj-0 hWRVZg"
>
   
  <svg>
    close.svg
  </svg>
</button>
`;
